lambdas/es/indexer/index.py

```diff
@@ -163,25 +163,35 @@ def send_all(self):
             id_to_doc = {d["_id"]: d for d in self.queue}
             send_again = []
             for error in errors:
-                handled = False
                 # only retry index call errors, not delete errors
                 if "index" in error:
                     inner = error["index"]
                     info = inner.get("error")
+                    doc = id_to_doc[inner["_id"]]
                     # because error.error might be a string *sigh*
                     if isinstance(info, dict):
                         if "mapper_parsing_exception" in info.get("type", ""):
                             print("mapper_parsing_exception", error, inner)
-                            doc = id_to_doc[inner["_id"]]
                             # clear out structured metadata and try again
                             doc["user_meta"] = doc["system"] = {}
-                            handled = True
-                            send_again.append(doc)
-                if not handled:
-                    print("unhandled indexer error:", error)
+                        else:
+                            print("unhandled indexer error:", error)
+                    # Always retry, regardless of whether we know to handle and clean the request
+                    # or not. This can catch temporary 403 on index write blocks and other
+                    # transcient issues.
+                    send_again.append(doc)
```

Comment:
If index is not in error we won't retry (but should). I _think_ in this context that index is the operation that failed but it could also fail on a delete. There were also some "_index" keys if I recall. Seeing all the errors in the logs will help; or if there's API docs on this (I couldn't find them in the Python wrapper, but maybe the REST API has them).

Reply:
this will retry, its handled by the else in line 183 right below